cfpb/sbl-frontend
# feat(Upload): Skeleton, API Request, Upload Status

closes #302 
closes #305 
closes #307 

## Changes
- chore(.env): utilizes the `SBL_REGTECH_FILING_URL` env
- chore(TypeScript): resolves TypeScript types  - FilingStatus, Submissions, Uploads
- chore(filingTypes): moves associated filingTypes to the `types/filingTypes`
- feat(Upload, Submission): Implemented `useGetSubmissionLatest` and `useUploadMutation` as well as associated Axios requests
- component(InlineStatus): Used for the Icon - Description components used in Upload
- **feat(Upload): Uploading `.csv` is functioning properly**
- feat(Upload): Fetching previous uploads upon load of an `Upload` page

## How to Test
1. Make sure `SBL_REGTECH_FILING_URL` is set properly in your `.env`
2. Run `yarn update`, reset all Docker volumes and containers, and set the user to have a LEI in Keycloak
3. Navigate to `http://localhost:8899/filing` and click on 'Upload' to navigate to the lei's specific Upload page
4. Upload a `.csv` file -- can use test .csv files from https://github.com/cfpb/regtech-data-validator/tree/main/tests/data
5. Notice Upload icon and content changes; inspect the Network tab to verify API request results

## Screenshot
<img width="549" alt="Screenshot 2024-03-29 at 6 30 25 AM" src="https://github.com/cfpb/sbl-frontend/assets/13324863/f693311b-e798-47ad-8dc0-788fde91b565">
